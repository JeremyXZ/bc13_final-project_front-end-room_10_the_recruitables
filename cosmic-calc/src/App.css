.navBarPageDiv {
  background-image: url(./images/NavBar.png);
  background-size: cover;
  width: 1005;
  background-repeat: no-repeat;
  background-position: bottom center;
  display: flex;
  justify-content: space-between;
  align-items: center;
  height: 15vh;
  z-index: 5;
  padding-top: 5%;
  padding-bottom: 10%;
  padding-left: 1%;
  padding-right: 1%;
}

.navButtonProfile {
  background-image: url(./images/Background_Buttons/ProfileButtonE.png);
  background-size: cover;
  background-repeat: no-repeat;
  background-position: center;
  background-color: rgba(0, 0, 255, 0);
  width: 10vw;
  height: 10vh;
  min-width: 150px;
  min-height: 100px;
  text-align: center;
  object-fit: fill;
}

.navButtonGame {
  background-image: url(./images/Background_Buttons/GameButtonE.png);
  background-size: contain;
  background-repeat: no-repeat;
  background-position: center;
  background-color: rgba(0, 0, 255, 0);
  width: 12vw;
  height: 12vh;
  min-width: 150px;
  min-height: 100px;
  object-fit: fill;
  text-align: center;
}

.navButtonProgress {
  background-image: url(./images/Background_Buttons/ProgressButtonE.png);
  background-size: contain;
  background-repeat: no-repeat;
  background-position: center;
  background-color: rgba(0, 0, 255, 0);
  width: 12vw;
  height: 12vh;
  min-width: 160px;
  min-height: 100px;
  text-align: center;
  object-fit: fill;
}

.navButtonProfileHighlighted {
  background-image: url(./images/Background_Buttons/profile_button_glow.png);

  background-size: cover;
  background-repeat: no-repeat;
  background-position: center;
  background-color: rgba(255, 251, 0, 0);
  width: 10vw;
  height: 10vh;
  min-width: 140px;
  min-height: 100px;
  text-align: center;
}

.navButtonGameHighlighted {
  background-image: url(./images/Background_Buttons/game_button_glow.png);

  background-size: cover;
  background-repeat: no-repeat;
  background-position: center;
  background-color: rgba(0, 0, 255, 0);
  width: 10vw;
  height: 10vh;
  min-width: 140px;
  min-height: 100px;
  text-align: center;
}

.navButtonProgressHighlighted {
  background-image: url(./images/Background_Buttons/progress_button_glow.png);

  background-size: cover;
  background-repeat: no-repeat;
  background-position: center;
  background-color: rgba(0, 0, 255, 0);
  width: 10vw;
  height: 10vh;
  min-width: 140px;
  min-height: 100px;
  text-align: center;
}

.navButtonLogin {
  background-image: url(./images/Background_Buttons/loginButton.png);
  background-size: cover;
  background-repeat: no-repeat;
  background-position: center;
  background-color: rgba(0, 0, 255, 0);
  width: 15vw;
  height: 10vh;
  min-width: 180px;
  min-height: 100px;
  text-align: center;
  color: rgba(0, 0, 255, 0);
}

.navButtonRegister {
  background-image: url(./images/Background_Buttons/registerButton.png);
  background-size: cover;
  background-repeat: no-repeat;
  background-position: center;
  background-color: rgba(34, 34, 170, 0);
  width: 15vw;
  height: 10vh;
  min-width: 180px;
  min-height: 100px;
  text-align: center;
  color: #ffffff00;
}

.navButtonProfile:hover,
.navButtonGame:hover,
.navButtonProgress:hover,
.navButtonRegister:hover,
.navButtonLogin:hover {
  transition: transform 0.3s ease-in-out 0s;
  transform: rotate(5deg) scale(1.2);
  cursor: pointer;
}

.navBarLoginDiv {
  display: flex;
  justify-content: space-between;
  align-items: center;
  /* width: 30vw; */
  height: 2cm;
  margin: 1cm 1cm;
}

.progress-score {
  width: 15vw;
  height: 8vh;
  display: flex;
  object-fit: scale-down;
  outline: #f70d11 5px groove;
  background-color: gold;
  justify-content: center;
  align-items: center;
  text-align: center;
  font-size: 0.9em;
  color: black;
  border-radius: 50%;
  /* text-shadow: #fff 1px 1px; */
  margin-top: 2vh;
  z-index: 10;
  padding: 0.7cm;
  animation: rotation 2s infinite linear;
  transition: transform 1s ease-in-out;
}

.logo {
  margin-top: 2%;
  object-fit: scale-down;
  overflow: visible;
  width: 15vw;
  height: 30vh;
  animation: rotation 2s infinite linear;
  transition: transform 1s ease-in-out;
}

.progress-score:hover {
  transition: 2s;
  transform: scale(1, 0.5);
}

.logo:hover {
  transition: 2s;
  transform: rotate(360deg);
}

@media screen and (max-width: 650px) {
  .navBarPageDiv {
   width: 110%;
  }
}

@media screen and (max-width: 380px) {
  .navBarPageDiv {
    width: 160%;
  }
}