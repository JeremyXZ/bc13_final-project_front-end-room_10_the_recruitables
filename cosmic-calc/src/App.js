import React, { useState, useContext } from "react";
import { Routes, Route, useNavigate, useLocation } from "react-router-dom";
import Profile from "./pages/Profile";
import Game from "./pages/Game";
import Progress from "./pages/Progress";
import Login from "./pages/LoginPage";
import Register from "./pages/RegisterPage";
import ProtectedRoute from "./components/Authentication/requireAuth";
import Logout from "./components/Authentication/Logout";
import "./App.css";
import { auth } from "./pages/firebaseConfig";
import { ScoreContext } from "./components/score/ScoreContext";
import TimedGame from "./pages/TimedGame";
import Avatars from "./pages/Avatars";
import Level1 from "./pages/Level1";
import Level5 from "./pages/Level5";
import useSound from "use-sound";
import buttonFX from "./components/sound/FX/buttonFX.mp3";
import YearTwo from "./pages/YearTwoGames";
import logo from "../src/images/Logo.png";
import YearThreeGames from "./pages/YearThreeGames";
import YearFourGames from "./pages/YearFourGames";


export default function App() {
  const authed = auth;
  let context = useContext(ScoreContext);
  const location = useLocation();
  const navigate = useNavigate();
  const [profileHighlighted, setProfileHighlighted] = useState(false);
  const [progressHighlighted, setProgressHighlighted] = useState(false);
  const [gameHighlighted, setGameHighlighted] = useState(false);
  const [playHover] = useSound(buttonFX, {
    volume: 0.3,
    playbackRate: Math.floor(Math.random() * (2 - 0.8) + 0.8),
  });

  const navigateToLogin = () => {
    navigate("/");
  };

  const navigateToProfile = () => {
    navigate("/profile");

    setProfileHighlighted(true);
    setProgressHighlighted(false);
    setGameHighlighted(false);
    context.update();
  };

  const navigateToProgress = () => {
    navigate("/progress");
    setProfileHighlighted(false);
    setProgressHighlighted(true);
    setGameHighlighted(false);
    context.update();
  };

  const navigateToGame = () => {
    navigate("/game");
    setProfileHighlighted(false);
    setProgressHighlighted(false);
    setGameHighlighted(true);
    context.update();
    console.log(navigate);
  };

  const navigateToTimedGame = () => {
    navigate("/timedGame");
  };

  const navigateToAvatars = () => {
    navigate("/avatars");
  };

  const navigateToLevel1 = () => {
    navigate("/level1");
  };

  const navigateToLevel5 = () => {
    navigate("/level5");
  };

  const navigateToRegister = () => {
    navigate("/register");
  };

  const navigateToYearTwo = () => {
    navigate("/year-two-games");
  };

  const navigateToYearThree = () => {
    navigate("/year-three-games");
  };

  const navigateToYearFour = () => {
    navigate("/year-four-games");
  };

  console.log(authed.currentUser);

  return (
    <div>
      {/* {authed.currentUser ? ( */}
        <div className="navBarPageDiv">
          <img src={logo} alt="logo" className="logo" />
          <button onClick={navigateToYearTwo} className="navButtonYearTwo">
            Year Two
          </button>
          <button onClick={navigateToYearThree} className="navButtonYearThree">
            Year Three
          </button>
          <button onClick={navigateToYearFour} className="navButtonYearFour">
            Year Four
          </button>
          {/* {location.pathname !== "/profile" && (
            <button
              onClick={navigateToProfile}
              onMouseOver={playHover}
              className={
                profileHighlighted
                  ? "navButtonProfileHighlighted"
                  : "navButtonProfile"
              }
            ></button>
          )}
          {location.pathname !== "/progress" && (
            <button
              onClick={navigateToProgress}
              onMouseOver={playHover}
              className={
                progressHighlighted
                  ? "navButtonProgressHighlighted"
                  : "navButtonProgress"
              }
            ></button>
          )}
          {location.pathname !== "/game" && (
            <button
              onClick={navigateToGame}
              onMouseOver={playHover}
              className={
                gameHighlighted ? "navButtonGameHighlighted" : "navButtonGame"
              }
            ></button>
          )}
          {location.pathname !== "/game" && (
            <div className="progress-score">
              <h1>Score: {context.score}</h1>
            </div>
          )} */}
          <Logout />
          {/* <button onClick={navigateToTimedGame} className="navButtonTimedGame">
            TimedGame
          </button>
          <button onClick={navigateToAvatars} className="navButtonAvatars">
            Avatars
          </button>
          <button onClick={navigateToLevel1} className="navButtonLevel1">
            Level1
          </button>
          <button onClick={navigateToLevel5} className="navButtonLevel5">
            Level5

          </button>
          <button onClick={navigateToYearTwo} className="navButtonLevel6">
            year-Two
          </button>
          </button> */}
        {/* </div> */}
      {/* ) : (
        <div className="navBarLoginDiv">
          {location.pathname !== "/" && (
            <button
              className="navButtonLogin"
              onMouseOver={playHover}
              onClick={navigateToLogin}
            >
              Login
            </button>
          )}
          {location.pathname !== "/register" && (
            <button
              className="navButtonRegister"
              onMouseOver={playHover}
              onClick={navigateToRegister}
            >
              Register
            </button>
          )} */}
        </div>
    

      <Routes>
        <Route path="/" element={<Login />} />
        <Route path="/register" element={<Register />} />
        <Route
          path="/profile"
          element={
            <ProtectedRoute>
              <Profile />
            </ProtectedRoute>
          }
        />
        <Route
          path="/progress"
          element={
            <ProtectedRoute>
              <Progress />
            </ProtectedRoute>
          }
        />
        <Route
          path="/year-two-games"
          element={
            <ProtectedRoute>
              <YearTwo />
            </ProtectedRoute>
          }
        />
        <Route
          path="/game"
          element={
            <ProtectedRoute>
              <Game />
            </ProtectedRoute>
          }
        />
        <Route
          path="/year-three-games"
          element={
            <ProtectedRoute>
              <YearThreeGames />
            </ProtectedRoute>
          }
        />
        <Route
          path="/year-four-games"
          element={
            <ProtectedRoute>
              <YearFourGames />
            </ProtectedRoute>
          }
        />
        <Route
          path="/timedGame"
          element={
            <ProtectedRoute>
              <TimedGame />
            </ProtectedRoute>
          }
        />
        <Route
          path="/avatars"
          element={
            <ProtectedRoute>
              <Avatars />
            </ProtectedRoute>
          }
        />
        <Route
          path="/level1"
          element={
            <ProtectedRoute>
              <Level1 />
            </ProtectedRoute>
          }
        />
        <Route
          path="/level5"
          element={
            <ProtectedRoute>
              <Level5 />
            </ProtectedRoute>
          }
        />
      </Routes>
    </div>
  );
}
