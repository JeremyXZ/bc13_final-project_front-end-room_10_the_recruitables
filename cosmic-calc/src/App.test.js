import {render, screen} from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import React from 'react'
import '@testing-library/jest-dom'
import App from './App';
import {BrowserRouter, MemoryRouter, useLocation} from 'react-router-dom'


describe('<App/> rendering and navigataing', () => {
    
    it("successfully renders the component, matches snapshot", () => {
        const { asFragment } = render(<App />, {wrapper: BrowserRouter});
        expect(asFragment()).toMatchSnapshot();
      });

    it("verifies page content for default route", async () => {
        render(<App />, {wrapper: BrowserRouter})              
        expect(screen.getByPlaceholderText(/email/i)).toBeInTheDocument()
        expect(screen.getByPlaceholderText(/password/i)).toBeInTheDocument()
        expect(screen.getByRole('button', {name: /login/i})).toBeInTheDocument()    
    
    })

    it('verifies page content for expected route after navigating- from home page to register page', async () => {
        render(<App />, {wrapper: BrowserRouter})   
        await userEvent.click(screen.getByRole('button', {name: /register/i}))
        expect(screen.getByPlaceholderText(/name/i)).toBeInTheDocument()
        expect(screen.getByPlaceholderText(/email/i)).toBeInTheDocument()
    })
})
