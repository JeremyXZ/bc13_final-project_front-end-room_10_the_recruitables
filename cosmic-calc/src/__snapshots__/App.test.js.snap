// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App/> rendering and navigataing successfully renders the component, matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="navBarPageDiv"
    >
      <img
        alt="logo"
        class="logo"
        src="Logo.png"
      />
      <button
        class="navButtonYearTwo"
      >
        Year Two
      </button>
      <button
        class="navButtonYearThree"
      >
        Year Three
      </button>
      <button
        class="navButtonYearFour"
      >
        Year Four
      </button>
      <button
        class="navButtonLogout"
      />
    </div>
    <div
      class="Login"
    >
      <div>
        <div
          class="loginLogoDiv"
        >
          <img
            alt="logo"
            class="loginLogo"
            src="Logo.png"
          />
        </div>
        <form>
          <div
            class="inputLoginDiv"
          >
            <input
              class="loginInput"
              placeholder="email"
              type="email"
              value=""
            />
            <input
              class="passwordInput"
              placeholder="password"
              type="password"
              value=""
            />
          </div>
          <div
            class="loginButtonDiv"
          >
            <button
              class="loginButton"
              type="submit"
            >
              Login
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
